go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -64,6 +66,9 @@ type CommonNetworkControllerInfo struct {
 // by more than one type of network controllers.
 type BaseNetworkController struct {
 	CommonNetworkControllerInfo
+	// per controller nad/netconf name information
+	util.NetInfo
+	util.NetConfInfo
```

Comment:
this is not common among different network controllers. why part of CommonNetworkControllerInfo? They are actually essential information for different network controller.

util.NetInfo is actually used for default network controller as well. (GetNetworkName, Prefix, IsSecondary check). I can probably implement NetConfInfo for default network as well, but most of its method would be no-op.

Reply:
I don't think anything that I said precludes anything of what you said.

All network controllers have to have `util.NetInfo` and `util.NetConfInfo` so it could be part `CommonNetworkControllerInfo`.

For the default network controller `util.NetInfo` is always nil, so it make sense for it to be a property instead of embed. Or alternatively have a default implementation of `util.NetInfo` for the default network controller.

Reply:
Oh, so I guess that you mean that all controllers should have the same value for CommonNetworkControllerInfo. I was talking about the type being common, but not the value.

Ok, let's leave it as part of `BaseNetworkController` then.

Point 2 still applies though. I don't think if should be embed for the reasons stated above. Which also goes inline with trying to avoid using the expression `(*util.NetNameInfo)(nil)` in several places around the code, it's somewhat ugly.